modules/gke-nodepool/main.tf

```diff
@@ -129,6 +128,15 @@ resource "google_container_node_pool" "nodepool" {
     }
   }
 
+  dynamic "placement_policy" {
+    for_each = try(var.nodepool_config.placement_policy, null) != null ? [""] : []
+    content {
+      type         = try(var.nodepool_config.placement_policy.type, null)
+      policy_name  = try(var.nodepool_config.placement_policy.policy_name, null)
+      tpu_topology = try(var.nodepool_config.placement_policy.tpu_topology, null)
```

Comment:
These `try`s are not needed, `for_each` guarantees existence of `var.nodepool_config.placement_policy`